chipsec/modules/debugenabled.py

```diff
@@ -0,0 +1,120 @@
+#
+# Main module functionality
+#
+########################################################################################################
+
+class debugenabled(chipsec.module_common.BaseModule):
+
+
+
+    def __init__(self):
+        BaseModule.__init__(self)
+
+    def is_supported(self):
+        current_platform_id = self.cs.get_chipset_id()
+        supported = (current_platform_id == chipsec.chipset.CHIPSET_ID_CFL) | (current_platform_id == chipsec.chipset.CHIPSET_ID_KBL) | (current_platform_id == chipsec.chipset.CHIPSET_ID_SKL)
+        if not supported: self.logger.log_skipped_check( "DCI is not supported in this platform" )
+        return supported
+
+
+    def check_dci( self ):
+        TestFail = False;
+        self.logger.log( "[X] Checking DCI register status" )
+        value = self.cs.msgbus.mm_msgbus_reg_read(P2SB_DCI_PORT_ID,DCI_CONTROL_REG_OFFSET)
+        if self.logger.VERBOSE: self.logger.log( '[*] DCI Control Register = 0x%X' % value )
+        HDCIEN = ((value & HDCIEN_MASK) == HDCIEN_MASK)
+        if HDCIEN:
+            TestFail = True;
+        return TestFail
+
+    def check_cpu_debug_enable( self ):
+        self.logger.log( "[X] Checking IA32_DEBUG_INTERFACE msr status" )
+        TestFail = False;
+        for tid in range(self.cs.msr.get_cpu_thread_count()):
+            (eax, edx) = self.cs.helper.read_msr( tid, IA32_DEBUG_INTERFACE_MSR )
+            if self.logger.VERBOSE: self.logger.log('[cpu%d] RDMSR( 0x%x ): EAX = 0x%08X, EDX = 0x%08X' % (tid, IA32_DEBUG_INTERFACE_MSR, eax, edx) )
+            IA32_DEBUG_INTERFACE_DEBUGENABLE = ((IA32_DEBUG_INTERFACE_DEBUGENABLE_MASK & eax) == IA32_DEBUG_INTERFACE_DEBUGENABLE_MASK)
+            IA32_DEBUG_INTERFACE_DEBUGELOCK = ((IA32_DEBUG_INTERFACE_DEBUGELOCK_MASK & eax) == IA32_DEBUG_INTERFACE_DEBUGELOCK_MASK)
+            IA32_DEBUG_INTERFACE_DEBUGEOCCURED = ((IA32_DEBUG_INTERFACE_DEBUGEOCCURED_MASK & eax) == IA32_DEBUG_INTERFACE_DEBUGEOCCURED_MASK)
+            if edx == EDX_ENABLE_STATE: #Sanity check only EAX matters
+                if (IA32_DEBUG_INTERFACE_DEBUGENABLE) or (IA32_DEBUG_INTERFACE_DEBUGELOCK) or (IA32_DEBUG_INTERFACE_DEBUGEOCCURED):
+                    TestFail = True;
+        return TestFail
+
+    def run( self, module_argv ):
+        if len(module_argv) > 2:
+            self.logger.error( 'Not expecting any arguments' )
+            return ModuleResult.ERROR
+        returned_result = ModuleResult.PASSED;
+        self.logger.start_test( "Debug features test" )
+        script_pa = None
+        dci_test_fail = self.check_dci();
+        cpu_debug_test_fail = self.check_cpu_debug_enable();
+            
+        if dci_test_fail == True:
+            self.logger.log_failed_check( 'DCI Debug is enabled' )
+            returned_result = ModuleResult.FAILED
+        if cpu_debug_test_fail == True:
+            self.logger.log_failed_check( 'CPU IA32_DEBUG_INTERFACE is enabled' )
+            returned_result = ModuleResult.FAILED
+        
```

Comment:
The log_failed_check should only be called once in a test.  Can you change this to log_failed for each of the blocks and then add a new entry below that calls log_failed_checked with a string like 'One or more debug interfaces enabled.' if returned_result is set to ModuelResult.FAILED.